hi-ml-azure/src/health/azure/azure_util.py

```diff
@@ -476,27 +556,148 @@ def get_latest_aml_runs_from_experiment(args: Namespace, workspace: Workspace) -
     return list(islice(experiment.get_runs(tags=tags), num_runs))
 
 
-def get_aml_runs_from_recovery_ids(args: Namespace, workspace: Workspace) -> List[Run]:
+def get_aml_runs_from_recovery_ids(args: Namespace, aml_workspace: Optional[Workspace] = None,
```

Comment:
there is code duplication - a funciton to get multiple runs should use "get single run" as a building block. 
Also, do we really need two arguments that differ only by the plural "s"? Wouldn't it be more pythonic to have run_recovery_id to accept single or multipe run IDs)

Reply:
That is what I originally intended to implement but then there were a few scenarios in which I wanted to call get_aml_run_from_id multiple times whilst iterating over run_ids, hence it made sense to me to create this wrapper function which performs that behavior (although granted this particular function does not call the singular version of the function so I can clean that up).

I also felt like it was clearer to have a separate argument for run_ids (plural) as it explicitly tells the user they can supply more than one id. If we just use the arg run_id we would still have to have a check at the start of each function whether it's a list of ids or a single one, which doesn't seem any neater to me. Happy to discuss further offline!

Reply:
Discussed offline - we will continue exploring this and will create a new PR if we do decide to refactor